### ISSUE TITLE
Any interest in cross platforming this to macOS?

### ISSUE THREAD

**Issue by lonkelle** [libhybris/libhybris#518]
Many attempts to build this for an M1 Mac failed so I was just curious if others had interest in using this on macOS or even iOS for that matter (loading Android libraries on iOS would be pretty dang cool).

**Comment by Thesola10** [comment]
The way I understand it, `libhybris` (this project) allows drivers written for the `bionic` C library (from Android) to run under the GNU C library (which most other Linux distributions use). It is intended to run non-Android Linux on Android devices, while benefiting from Android-specific drivers written by vendors.

What you are asking is fundamentally impossible as-is. The way Windows Subsystem for Android works is by spinning up a Linux virtual machine, and running an Android system (like Android-x86) on top of that.

The only operating system capable of running the Android runtime in its current state is Linux. Android relies heavily on Linux-specific behaviour, such as the custom-made `ashmem` and `binder` kernel modules, or the SELinux security framework.

With `libhybris`, we can run `glibc`-based Linux on Android-based Linux platforms, and with [Waydroid](https://github.com/Waydroid/Waydroid), we can run the Android environment within a `glibc`-based Linux (but even that requires enabling or installing Android's custom modules).

The only other OS that might be able to natively run the Android runtime is Google's own Fuchsia, but we have yet to figure out what they want to do with it.


**TL;DR**: _Android is Linux, and this library only allows two types of Linux "systems" to talk to each other. WSA only exists because it's a virtual machine under the hood. If you want to run Android apps on macOS, run a VM._

The state of pure AArch64 Android VM images is apparently [pretty bad](https://github.com/google/android-emulator-container-scripts/issues/192), so I can only recommend running an AArch64 VM of, say, Ubuntu, and installing Waydroid on it.

Or maybe dual-boot Asahi idk.
